models/Asset/Image/Thumbnail/Processor.php

```diff
@@ -143,7 +149,12 @@ public static function process(Asset $asset, Config $config, $fileSystemPath = n
 
         $image = Asset\Image::getImageTransformInstance();
 
-        if ($contentOptimizedFormat && self::hasWebpSupport() && $image->supportsFormat('webp')) {
+        if (self::hasWebpSupport() && $image->supportsFormat('webp')) {
```

Comment:
```suggestion
        if ($optimizedFormat && self::hasWebpSupport() && $image->supportsFormat('webp')) {
```
WebP is only used if format is set to `auto` (`$format == 'source'`), otherwise the defined target format is used.